### PR TITLE
single pool: authorized withdrawer fix

### PR DESCRIPTION
**Pull request by 2501babe**
ht to zellic. rebase after #4117 lands

### CODE REVIEW

**Review by joncinque — COMMENTED** [review]
The last commit here looks good. Might be worth adding tests trying to update the metadata with the voter and node keys

**Comment by 2501babe** [comment]
only new material is tests

File: stake-pool/single-pool/tests/update_pool_token_metadata.rs [hunk 1]
```diff
@@ -56,5 +57,79 @@ async fn success_update_pool_token_metadata() {
...
+
+    let mut instruction = instruction::update_token_metadata(
+        &id(),
+        &accounts.vote_account.pubkey(),
+        &accounts.withdrawer.pubkey(),
+        UPDATED_NAME.to_string(),
+        UPDATED_SYMBOL.to_string(),
+        UPDATED_URI.to_string(),
+    );
+    assert_eq!(instruction.accounts[2].pubkey, accounts.withdrawer.pubkey());
```
Comment (joncinque) [inline stake-pool/single-pool/tests/update_pool_token_metadata.rs:74]: Clever, makes me realize I should do this more often in tests

**Review by joncinque — APPROVED** [review]
Looks great!

**Pull request merged**